File: rust/src/api/streaming_stream.rs
// this file is @generated
use crate::{error::Result, models::*, Configuration};

#[derive(Default)]
pub struct StreamingStreamListOptions {
    /// Limit the number of returned items
    pub limit: Option<i32>,

    /// The iterator returned from a prior invocation
    pub iterator: Option<String>,

    /// The sorting order of the returned items
    pub order: Option<Ordering>,
}

#[derive(Default)]
pub struct StreamingStreamCreateOptions {
    pub idempotency_key: Option<String>,
}

pub struct StreamingStream<'a> {
    cfg: &'a Configuration,
}

impl<'a> StreamingStream<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self {
            cfg,
        }
    }

    /// List of all the organization's streams.
    pub async fn list(
        &self,
        options: Option<StreamingStreamListOptions>,
    ) -> Result<ListResponseStreamOut> {
        let StreamingStreamListOptions {
            limit,
            iterator,
            order,
        } = options.unwrap_or_default();

        crate::request::Request::new(
            http1::Method::GET,
            "/api/v1/stream",
        )
            .with_optional_query_param("limit", limit)
            .with_optional_query_param("iterator", iterator)
            .with_optional_query_param("order", order)
            .execute(self.cfg)
            .await
    }

    /// Creates a new stream.
    pub async fn create(
        &self,
        stream_in: StreamIn,
        options: Option<StreamingStreamCreateOptions>,
    ) -> Result<StreamOut> {
        let StreamingStreamCreateOptions {
            idempotency_key,
        } = options.unwrap_or_default();

        crate::request::Request::new(
            http1::Method::POST,
            "/api/v1/stream",
        )
            .with_optional_header_param("idempotency-key", idempotency_key)
            .with_body_param(stream_in)
            .execute(self.cfg)
            .await
    }

    /// Get a stream by id or uid.
    pub async fn get(
        &self,
        stream_id: String,
    ) -> Result<StreamOut> {
        crate::request::Request::new(
            http1::Method::GET,
            "/api/v1/stream/{stream_id}",
        )
            .with_path_param("stream_id", stream_id)
            .execute(self.cfg)
            .await
    }

    /// Update a stream.
    pub async fn update(
        &self,
        stream_id: String,
        stream_in: StreamIn,
    ) -> Result<StreamOut> {
        crate::request::Request::new(
            http1::Method::PUT,
            "/api/v1/stream/{stream_id}",
        )
            .with_path_param("stream_id", stream_id)
            .with_body_param(stream_in)
            .execute(self.cfg)
            .await
    }

    /// Delete a stream.
    pub async fn delete(
        &self,
        stream_id: String,
    ) -> Result<()> {
        crate::request::Request::new(
            http1::Method::DELETE,
            "/api/v1/stream/{stream_id}",
        )
            .with_path_param("stream_id", stream_id)
            .returns_nothing()
            .execute(self.cfg)
            .await
    }

    /// Partially update a stream.
    pub async fn patch(
        &self,
        stream_id: String,
        stream_patch: StreamPatch,
    ) -> Result<StreamOut> {
        crate::request::Request::new(
            http1::Method::PATCH,
            "/api/v1/stream/{stream_id}",
        )
            .with_path_param("stream_id", stream_id)
            .with_body_param(stream_patch)
            .execute(self.cfg)
            .await
    }
}

File: javascript/src/api/streamingStream.ts
// this file is @generated

import {
  type ListResponseStreamOut,
  ListResponseStreamOutSerializer,
} from "../models/listResponseStreamOut";
import type { Ordering } from "../models/ordering";
import { type StreamIn, StreamInSerializer } from "../models/streamIn";
import { type StreamOut, StreamOutSerializer } from "../models/streamOut";
import { type StreamPatch, StreamPatchSerializer } from "../models/streamPatch";
import { HttpMethod, SvixRequest, type SvixRequestContext } from "../request";

export interface StreamingStreamListOptions {
  /** Limit the number of returned items */
  limit?: number;
  /** The iterator returned from a prior invocation */
  iterator?: string | null;
  /** The sorting order of the returned items */
  order?: Ordering;
}

export interface StreamingStreamCreateOptions {
  idempotencyKey?: string;
}

export class StreamingStream {
  public constructor(private readonly requestCtx: SvixRequestContext) {}

  /** List of all the organization's streams. */
  public async list(
    options?: StreamingStreamListOptions
  ): Promise<ListResponseStreamOut> {
    const request = new SvixRequest(HttpMethod.GET, "/api/v1/stream");

    request.setQueryParams({
      limit: options?.limit,
      iterator: options?.iterator,
      order: options?.order,
    });

    return await request.send(
      this.requestCtx,
      ListResponseStreamOutSerializer._fromJsonObject
    );
  }

  /** Creates a new stream. */
  public async create(
    streamIn: StreamIn,
    options?: StreamingStreamCreateOptions
  ): Promise<StreamOut> {
    const request = new SvixRequest(HttpMethod.POST, "/api/v1/stream");

    request.setHeaderParam("idempotency-key", options?.idempotencyKey);
    request.setBody(StreamInSerializer._toJsonObject(streamIn));

    return await request.send(this.requestCtx, StreamOutSerializer._fromJsonObject);
  }

  /** Get a stream by id or uid. */
  public async get(streamId: string): Promise<StreamOut> {
    const request = new SvixRequest(HttpMethod.GET, "/api/v1/stream/{stream_id}");

    request.setPathParam("stream_id", streamId);

    return await request.send(this.requestCtx, StreamOutSerializer._fromJsonObject);
  }

  /** Update a stream. */
  public async update(streamId: string, streamIn: StreamIn): Promise<StreamOut> {
    const request = new SvixRequest(HttpMethod.PUT, "/api/v1/stream/{stream_id}");

    request.setPathParam("stream_id", streamId);
    request.setBody(StreamInSerializer._toJsonObject(streamIn));

    return await request.send(this.requestCtx, StreamOutSerializer._fromJsonObject);
  }

  /** Delete a stream. */
  public async delete(streamId: string): Promise<void> {
    const request = new SvixRequest(HttpMethod.DELETE, "/api/v1/stream/{stream_id}");

    request.setPathParam("stream_id", streamId);

    return await request.sendNoResponseBody(this.requestCtx);
  }

  /** Partially update a stream. */
  public async patch(streamId: string, streamPatch: StreamPatch): Promise<StreamOut> {
    const request = new SvixRequest(HttpMethod.PATCH, "/api/v1/stream/{stream_id}");

    request.setPathParam("stream_id", streamId);
    request.setBody(StreamPatchSerializer._toJsonObject(streamPatch));

    return await request.send(this.requestCtx, StreamOutSerializer._fromJsonObject);
  }
}
